### PR TITLE
Adding new interface ITestDiscoveryEventHandler2.

### PR DESCRIPTION
**Pull request by khilangoel**
Adding new interface ITestDiscoveryEventHandler2. This PR doesnt include scenarios for IDE. Will add that in next PR .

### CODE REVIEW

File: src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/DiscoveryCompleteEventsHandler.cs
Comment (khilan2goel) [inline src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/DiscoveryCompleteEventsHandler.cs]: This class will be used to handle requests from IDE. Will be used later.

File: src/Microsoft.TestPlatform.ObjectModel/Client/Interfaces/ITestDiscoveryEventsHandler2.cs
Comment (harshjain2) [inline src/Microsoft.TestPlatform.ObjectModel/Client/Interfaces/ITestDiscoveryEventsHandler2.cs]: Could have inherited from `ITestDiscoveryEventsHandler' added `void HandleDiscoveryComplete(DiscoveryCompleteEventArgs discoveryCompleteEventArgs, IEnumerable<TestCase> lastChunk);`

Comment (khilan2goel) [reply src/Microsoft.TestPlatform.ObjectModel/Client/Interfaces/ITestDiscoveryEventsHandler2.cs]: We will be exposing this Interface to users. In inheriting from old interface case, users have to implement all the methods. So, that's why didn't go with that approach.  I have already written converter so that existing clients do not break which I will raise in next PR.

File: src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/DiscoveryCompleteEventsHandler.cs
Comment (harshjain2) [inline src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/DiscoveryCompleteEventsHandler.cs]: Suggestion : Rename `DiscoveryCompleteEventsHandler` -> `TestDiscoveryEventsHandler` or `DiscoveryEventsHandler`

File: src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/DiscoveryCompleteEventsHandler.cs
Comment (harshjain2) [inline src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/DiscoveryCompleteEventsHandler.cs]: Suggestion : Add note that `Telemetry` can be collected from here.

File: src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/DiscoveryCompleteEventsHandler.cs
Comment (harshjain2) [inline src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/DiscoveryCompleteEventsHandler.cs]: Suggestion : We have implemented a new interface `ITestDiscoveryEventsHandler2`, should the clients just move to new implementation, when ever they upgrade to latest version of TestPlatform?

File: src/testhost.x86/testhost.x86.csproj
Comment (harshjain2) [inline src/testhost.x86/testhost.x86.csproj]: 👍

File: test/Microsoft.TestPlatform.Client.UnitTests/Discovery/DiscoveryRequestTests.cs
Comment (harshjain2) [inline test/Microsoft.TestPlatform.Client.UnitTests/Discovery/DiscoveryRequestTests.cs]: Reorder alphabetically.

File: test/Microsoft.TestPlatform.CommunicationUtilities.UnitTests/ObjectModel/TestDiscoveryEventHandlerTests.cs
Comment (harshjain2) [inline test/Microsoft.TestPlatform.CommunicationUtilities.UnitTests/ObjectModel/TestDiscoveryEventHandlerTests.cs]: 👍

File: src/Microsoft.TestPlatform.Common/Interfaces/Engine/TesthostProtocol/IDiscoveryManager.cs
Comment (harshjain2) [inline src/Microsoft.TestPlatform.Common/Interfaces/Engine/TesthostProtocol/IDiscoveryManager.cs]: nit : remove extra line.

File: src/Microsoft.TestPlatform.CommunicationUtilities/EventHandlers/TestDiscoveryEventHandler.cs
Comment (harshjain2) [inline src/Microsoft.TestPlatform.CommunicationUtilities/EventHandlers/TestDiscoveryEventHandler.cs]:
```csharp
if(EqtTrace.IsInfoEnabled)
{
EqtTrace.Info(discoveryCompleteEventArgs.IsAborted ? "Discover Aborted." : "Discover Finished.");
}
```

**Review by harshjain2 — COMMENTED** [review]
Please address comments.

File: src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/DiscoveryCompleteEventsHandler.cs
Comment (harshjain2) [reply src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/DiscoveryCompleteEventsHandler.cs]: Discussed offline. This is resolved.

File: src/Microsoft.TestPlatform.Client/Discovery/DiscoveryRequest.cs
Comment (harshjain2) [inline src/Microsoft.TestPlatform.Client/Discovery/DiscoveryRequest.cs]: nit : ordering of namespaces.

File: src/Microsoft.TestPlatform.Common/Interfaces/Engine/TesthostProtocol/IDiscoveryManager.cs
Comment (harshjain2) [inline src/Microsoft.TestPlatform.Common/Interfaces/Engine/TesthostProtocol/IDiscoveryManager.cs]: nit : ordering of namespaces.

File: src/Microsoft.TestPlatform.CommunicationUtilities/EventHandlers/TestDiscoveryEventHandler.cs
Comment (harshjain2) [inline src/Microsoft.TestPlatform.CommunicationUtilities/EventHandlers/TestDiscoveryEventHandler.cs]: nit : add a space above this line.

File: src/Microsoft.TestPlatform.Client/Discovery/DiscoveryRequest.cs
Comment (harshjain2) [inline src/Microsoft.TestPlatform.Client/Discovery/DiscoveryRequest.cs]: nit : `ITestDiscoveryEventsHandler` -> `ITestDiscoveryEventsHandler2`

File: src/Microsoft.TestPlatform.ObjectModel/Client/Interfaces/ITestDiscoveryEventsHandler2.cs
Comment (harshjain2) [inline src/Microsoft.TestPlatform.ObjectModel/Client/Interfaces/ITestDiscoveryEventsHandler2.cs]: Interface `ITestDiscoveryEventsHander` existed under `Microsoft.VisualStudio.TestPlatform.ObjectModel.Client` namespace. 

Should we use the same namespace as used by v1 interface?

File: src/Microsoft.TestPlatform.CrossPlatEngine/Client/InProcessProxyDiscoveryManager.cs
Comment (harshjain2) [inline src/Microsoft.TestPlatform.CrossPlatEngine/Client/InProcessProxyDiscoveryManager.cs]: nit : reordering of namespaces.

File: src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelDiscoveryEventsHandler.cs
Comment (harshjain2) [inline src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelDiscoveryEventsHandler.cs]: nit : ordering of namespaces.

File: src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelProxyDiscoveryManager.cs
Comment (harshjain2) [inline src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelProxyDiscoveryManager.cs]: nit : ordering of namespaces.

File: src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyDiscoveryManager.cs
Comment (harshjain2) [inline src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyDiscoveryManager.cs]: nit: ordering of namespaces.

File: test/TranslationLayer.UnitTests/DiscoveryCompleteEventHandlerTests.cs
Comment (harshjain2) [inline test/TranslationLayer.UnitTests/DiscoveryCompleteEventHandlerTests.cs]: Rename class and file name.

File: test/TranslationLayer.UnitTests/DiscoveryCompleteEventHandlerTests.cs
Comment (harshjain2) [inline test/TranslationLayer.UnitTests/DiscoveryCompleteEventHandlerTests.cs]: HandleDiscoveryTestShouldCallTestDiscoveryHandler1Method -> HandleDiscoveredTestsShouldCallTestDiscoveryHandler1Method

**Review by harshjain2 — APPROVED** [review]
Approved with suggestions.

Please address all the comments and wait for one more reviewer to approve before merging.

**Comment by singhsarab** [comment]
> ��<?xml version="1.0" encoding="utf-16"?>

Please check for this change.

---
Refers to: src/Microsoft.TestPlatform.ObjectModel/Microsoft.TestPlatform.ObjectModel.csproj:1 in cba09ca. [](commit_id = cba09cac2d5b95e3c76af875a9ca94998bc59971, deletion_comment = False)

File: src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/DiscoveryEventsHandler.cs
Comment (singhsarab) [inline src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/DiscoveryEventsHandler.cs]: We are not using this anywhere. Please take this separately while enabling changes for Translation Layer.

**Review by singhsarab — APPROVED** [review]
Looks good with suggestions.

**Pull request merged**
